src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+                                          backendInfo: CallBackendInfo): Future[Unit] = {
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      executionQuery = dataAccess.executions.filter(execution =>
+        execution.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          execution.callFqn === call.fullyQualifiedName)
+
+      executionUpdate <- executionQuery.map(_.status).update(backendInfo.status.toString)
+
+      _ = require(executionUpdate == 1, s"Unexpected execution update count $executionUpdate")
+
+      executionResult <- executionQuery.result.head
+
+      backendUpdate <- backendInfo match {
+        case localBackendInfo: LocalCallBackendInfo =>
+          dataAccess.localJobs.filter(
+            _.executionId === executionResult.executionId
+          ).map(cols => (
+            cols.pid,
+            cols.rc
+            )).update(
+              localBackendInfo.processId,
+              localBackendInfo.resultCode
+            )
+
+        case jesBackendInfo: JesCallBackendInfo =>
+          dataAccess.jesJobs.filter(
+            _.executionId === executionResult.executionId
+          ).map(cols => (
+            cols.jesId,
+            cols.jesStatus
+            )).update(
+              jesBackendInfo.jesId,
+              jesBackendInfo.jesStatus
+            )
+
+        case null =>
+          throw new NotImplementedError("Backend info is null")
+      }
+
+      _ = require(backendUpdate == 1, s"Unexpected backend update count $backendUpdate")
+
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  /** Get all inputs for the scope of this call. */
+  override def getInputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      symbolResults <- dataAccess.symbols.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          symbol.io === IoInput &&
+          symbol.scope === call.fullyQualifiedName).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
+        new SymbolStoreEntry(
+          symbolStoreKey,
+          wdlType,
+          Option(symbolResult.wdlValue).map(wdlType.coerceRawValue(_).get))
+
+      }
+    } yield symbolStoreEntries
+
+    runTransaction(action)
+  }
+
+  /** Get all outputs for the scope of this call. */
+  override def getOutputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
```

Comment:
This looks basically the same as `getInputs` above, it seems this could be refactored into one common method parameterized by inputs/outputs.